states/visited-new.go

```diff
@@ -48,6 +51,11 @@ func (vc *VisitedCollection2) Add(ctx context.Context, target domain.Target, pla
 	sts, found := vc.visited[newKey]
 	if found {
 		vc.mu.Unlock()
+		if !sts.targetInput.Equals(newSts.targetInput) {
+			nb, _ := json.Marshal(newSts.targetInput)
+			b, _ := json.Marshal(sts.targetInput)
+			fmt.Println(fmt.Sprintf("%s %s\ncreated:%s\nfound:  %s", target.String(), strings.Join(overridingVars.Sorted(), " "), string(nb), string(b)))
+		}
```

Comment:
Just a reminder to remove this debug output. It's expected that the same hashes don't necessarily match completely, due to the `DefaultValue` field, which we want to ignore anyway.